libsolidity/formal/SMTEncoder.cpp

```diff
@@ -1018,6 +1017,22 @@ void SMTEncoder::visitFunctionIdentifier(Identifier const& _identifier)
 	}
 }
 
+void SMTEncoder::visitStructConstructorCall(FunctionCall const& _funCall) {
+	solAssert(*_funCall.annotation().kind == FunctionCallKind::StructConstructorCall, "");
+	auto structSymbolicVar = dynamic_pointer_cast<smt::SymbolicStructVariable>(m_context.expression(_funCall));
```

Comment:
```suggestion
	auto const& structSymbolicVar = dynamic_cast<smt::SymbolicStructVariable const&>(*m_context.expression(_funCall));
```
Then the assert below can be removed.